apps/opik-backend/src/main/java/com/comet/opik/infrastructure/EncryptionUtils.java

```diff
@@ -16,29 +14,17 @@
 
 public class EncryptionUtils {
 
-    private static void init() {
-        if (key != null) return;
-        synchronized (EncryptionUtils.class) {
-            if (key == null) {
-                byte[] keyBytes = config.getEncryption().getKey().getBytes(StandardCharsets.UTF_8);
-                key = new SecretKeySpec(keyBytes, ALGO);
-            }
-        }
-    }
-
     private static final String ALGO = "AES";
     private static final Base64.Encoder mimeEncoder = Base64.getMimeEncoder();
     private static final Base64.Decoder mimeDecoder = Base64.getMimeDecoder();
-    private static OpikConfiguration config;
     private static Key key;
 
-    @Inject
-    static void setConfig(OpikConfiguration config) {
-        EncryptionUtils.config = config;
+    public static void setConfig(OpikConfiguration config) {
+        byte[] keyBytes = config.getEncryption().getKey().getBytes(StandardCharsets.UTF_8);
+        key = new SecretKeySpec(keyBytes, ALGO);
     }
 
     public static String encrypt(String data) {
-        init();
         try {
```

Comment:
As `key` is now nullable, I suggest validating it's not null to avoid NPE

Reply:
@thiagohora I added annotations to public methods. But it's not related to what Ido is concerned. `key ` is a private field and is not a part of public api.